flink-clients/src/main/java/org/apache/flink/client/cli/ProgramOptions.java

```diff
@@ -119,6 +124,14 @@ else if (args.length > 0) {
 		} else {
 			this.savepointSettings = SavepointRestoreSettings.none();
 		}
+
+		if (line.hasOption(JOB_ID_OPTION.getOpt())) {
+			String jobIdRaw = line.getOptionValue(JOB_ID_OPTION.getOpt());
+			UUID id = UUID.fromString(jobIdRaw);
```

Comment:
If you are doing this in an effort to validate it, you should convert the exception into a `CliArgsException`